File: Timofeev-PE-42/Bizzare Underground C++/Project1/Box.cpp
#include "Box.h"
#include <iostream>

Box::Box() : isHanged(false), speedFalling(2), speedMovement(3) {
    std::cout << "--- Box Initialized ---" << std::endl;
    std::cout << "  Is Hanged: " << std::boolalpha << getIsHanged() << std::endl;
    std::cout << "  Speed Falling: " << getSpeedFalling() << std::endl;
    std::cout << "  Speed Movement: " << getSpeedMovement() << std::endl;
}

Box::~Box() {
    std::cout << "--- Box Destroyed ---" << std::endl;
}

void Box::checkHanged() {
    std::cout << "Box::checkHanged() called. Checking if box is hanging." << std::endl;
}

void Box::fallingBox() {
    std::cout << "Box::fallingBox() called. Simulating fall." << std::endl;
    std::cout << "Box::fallingBox(): Box is falling. (Requires GameObject positionY)" << std::endl;
}

void Box::movementBox() {
    std::cout << "Box::movementBox() called. Simulating movement." << std::endl;
}

bool Box::characterBoxContact(int characterPositionX, int characterPositionY) {
    std::cout << "Box::characterBoxContact() called. Checking Character(" << characterPositionX << ", " << characterPositionY << ") against Box." << std::endl;
    return false;
}

bool Box::getIsHanged() const { return isHanged; }
int Box::getSpeedFalling() const { return speedFalling; }
int Box::getSpeedMovement() const { return speedMovement; }

void Box::setIsHanged(bool hanged) {
    isHanged = hanged;
    std::cout << "Box::setIsHanged() called. New state: isHanged = " << std::boolalpha << isHanged << std::endl;
}
void Box::setSpeedFalling(int sf) {
    speedFalling = sf;
    std::cout << "Box::setSpeedFalling() called. New speed: " << speedFalling << std::endl;
}
void Box::setSpeedMovement(int sm) {
    speedMovement = sm;
    std::cout << "Box::setSpeedMovement() called. New speed: " << speedMovement << std::endl;
}

File: Timofeev-PE-42/Bizzare-Underground-Java/Box.java
public class Box { // Не наследуется от GameObject в этом примере
    private boolean isHanged;
    private int speedFalling;
    private int speedMovement;

    public Box() {
        this.isHanged = false;
        this.speedFalling = 2;
        this.speedMovement = 3;
        System.out.println("--- Box Initialized ---");
        System.out.println("  Is Hanged: " + isHanged);
        System.out.println("  Speed Falling: " + speedFalling);
        System.out.println("  Speed Movement: " + speedMovement);
    }

    @Override
    protected void finalize() throws Throwable {
        System.out.println("--- Box Destroyed ---");
        super.finalize();
    }

    public void checkHanged() {
        System.out.println("Box.checkHanged() called. Checking if box is hanging.");
        // Логика проверки
    }

    public void fallingBox() {
        System.out.println("Box.fallingBox() called. Simulating fall.");
        System.out.println("Box.fallingBox(): Box is falling. (Requires positionY)");
    }

    public void movementBox() {
        System.out.println("Box.movementBox() called. Simulating movement.");
    }

    public boolean characterBoxContact(int characterPositionX, int characterPositionY) {
        System.out.println("Box.characterBoxContact() called. Checking Character(" + characterPositionX + ", " + characterPositionY + ") against Box.");
        return false;
    }

    // Getters
    public boolean getIsHanged() { return isHanged; }
    public int getSpeedFalling() { return speedFalling; }
    public int getSpeedMovement() { return speedMovement; }

    // Setters
    public void setIsHanged(boolean hanged) { this.isHanged = hanged; }
    public void setSpeedFalling(int sf) { this.speedFalling = sf; }
    public void setSpeedMovement(int sm) { this.speedMovement = sm; }
}